packages/nimble-components/src/table/specs/data-hierarchy-hld.md

```diff
@@ -59,6 +53,69 @@ interface TableRowExpansionToggleEventDetail {
 
 Note: This event will _not_ be emitted for group rows.
 
+### `Table` API for specifying lazy loaded hierarchical data
+
+In some cases, a client might know that a given row has children, but for performance reasons, that data hasn't been loaded from the backend and given to the table. To enable this use case, there will be an API on the table to specify that a given row should be expandable even if it has no children within the data and to specify that a row is loading its children. That API will look as follows:
+
+```ts
+interface TableRowHierarchyOptions {
+    // The state of the row with respect to delayed hierarchy.
+    //
+    // The state can be one of the following:
+    // 1. none - There is no additional data about the hierarchy of the row beyond what is included in the table's data.
+    // 2. canLoadChildren - The row has children that haven't been loaded.
+    // 3. loadingChildren - The client application is loading the children of the row.
+    delayedHierarchyState: TableRowDelayedHierarchyState ;
+}
+
+export const TableRowDelayedHierarchyState = {
+    none: undefined,
+    canLoadChildren: 'canLoadChildren',
+    loadingChildren: 'loadingChildren'
+} as const;
+export type TableRowDelayedHierarchyState =
+    (typeof TableRowDelayedHierarchyState)[keyof typeof TableRowDelayedHierarchyState];
+
+public Table() {
+    ...
+    // Sets the hierarchy options for the rows specified by the passed IDs.
+    public async setRowHierarchyOptions(rowHierarchyOptions: { id: string, options: TableRowHierarchyOptions }[]): Promise<void>;
+}
+```
+
+The loading state of a row will look as follows:
+
+![Lazy Loading Spinner](./spec-images/LazyLoadingSpinner.gif)
+
+Some notes about the `setRowHierarchyOptions` API:
+
+-   previously set options will be cleared when `setData` is called
+-   previously set options will be cleared when the table's `idFieldName` changes
+-   previously set options will be cleared when the table's `parentIdFieldName` changes
+-   an option associated with a row ID that does not match a record in the data will be ignored
+-   all options will be ignored if `parentIdFieldName` is not configured on the table
+-   calling `setRowHierarchyOptions` when options have already been set on the table will have the following behavior:
+    -   existing options not passed in the new `rowHierarchyOptions` array will be left unmodified
+    -   existing options passed in the new `rowHierarchyOptions` array will be overwritten
```

Comment:
chatted with molly and meyers about aligning behavior closer to the setSelectedRecordIds api which will be updated here (not clearing on setData)